File: frontend/src/types/student.ts
export interface Student {
    id: string;
    namaDepan: string;
    namaBelakang?: string;
    tanggalLahir: string;
    usia: number;
}

export interface CreateStudentRequest {
    id: string;
    namaDepan: string;
    namaBelakang?: string;
    tanggalLahir: string;
}

export interface UpdateStudentRequest {
    id: string;
    namaDepan: string;
    namaBelakang?: string;
    tanggalLahir: string;
}

File: backend/CollegeStudent.Domain/Entities/Student.cs

namespace CollegeStudent.Domain.Entities;

public class Student
{
    public string Id { get; set; } =  string.Empty;
    public  string NamaDepan { get; set; } =  string.Empty;

    public string? NamaBelakang { get; set; }

    public  DateTime TanggalLahir { get; set;}

    public int Usia => (int)Math.Floor((DateTime.UtcNow - TanggalLahir). TotalDays / 365.25 );

}
